devito/ops/transformer.py

```diff
@@ -191,13 +190,18 @@ def create_ops_dat(f, name_to_ops_dat, block):
             )
         )))
 
-    res.append(Expression(ClusterizedEq(Eq(dim, ListInitializer(dim_shape)))))
-    res.append(Expression(ClusterizedEq(Eq(base, ListInitializer(base_val)))))
-    res.append(Expression(ClusterizedEq(Eq(d_p, ListInitializer(d_p_val)))))
-    res.append(Expression(ClusterizedEq(Eq(d_m, ListInitializer(d_m_val)))))
-    res.append(ops_decl_dat)
+    dim_val = Expression(ClusterizedEq(Eq(dim, ListInitializer(dim_val))))
+    base_val = Expression(ClusterizedEq(Eq(base, ListInitializer(base_val))))
+    d_p_val = Expression(ClusterizedEq(Eq(d_p, ListInitializer(d_p_val))))
+    d_m_val = Expression(ClusterizedEq(Eq(d_m, ListInitializer(d_m_val))))
 
-    return res
+    ops_dat = OpsDatDecl(dim_val=dim_val,
+                         base_val=base_val,
+                         d_p_val=d_p_val,
+                         d_m_val=d_m_val,
+                         ops_decl_dat=ops_decl_dat)
+
+    return ops_dat
```

Comment:
What do you about just returning the `OpsDatDecl`? 
The code sure is cleaner now.

Reply:
sure, thought about it as soon as pushed. thanks for pointing